### ISSUE TITLE
Scroll NaN rounding bug, desktop version

### ISSUE THREAD

**Issue by LAP2** [JetBrains/compose-jb#304]
Hi folks.

There is a divide by zero bug in androidx.compose.foundation.SliderAdapter class
During scrollScale calculation it is possible that contentSize would be equal to containerSize, as result we get NaN in scrollScale and "Cannot round NaN value." exception in the end.

Steps to reproduce:
1. Create window with vertical scroll, but larger that scrollable content, vertical scroll should not be visible.
2. Click in place where vertical scroll should be.

Scroll scale calculation code fragment:
`private val scrollScale get() = (containerSize - size) / (contentSize - containerSize)`

You can find exception text in attachment.
[NaN_exception.txt](https://github.com/JetBrains/compose-jb/files/5856555/NaN_exception.txt)

**Comment by dzirbel** [comment]
I'm seeing the same thing, sadly it freezes the entire UI with no explanation to the user. My strack trace is a little different, posting it here in case it's helpful:

<details>
<summary>Stack trace</summary>

```
Exception in thread "AWT-EventQueue-0 @coroutine#63641" java.lang.IllegalArgumentException: Cannot round NaN value.
	at kotlin.math.MathKt__MathJVMKt.roundToInt(MathJVM.kt:1132)
	at androidx.compose.foundation.ScrollableScrollbarAdapter.scrollTo(Scrollbar.desktop.kt:369)
	at androidx.compose.foundation.Scrollbar_desktopKt$scrollOnPressOutsideSlider$1$1.invokeSuspend(Scrollbar.desktop.kt:283)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at androidx.compose.ui.platform.FlushCoroutineDispatcher.flush(FlushCoroutineDispatcher.desktop.kt:56)
	at androidx.compose.ui.platform.FlushCoroutineDispatcher$dispatch$1$1.invokeSuspend(FlushCoroutineDispatcher.desktop.kt:40)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:316)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)

```
</details>

Compose 0.4.0-build182, JDK 16, Ubuntu 20.10.

**Comment by J-MR-T** [comment]
I am also experiencing this issue. From what I can see, this is triggered by the position of the sliderAdapter being NaN in Scrollbar.desktop.kt at line 296 (compose 0.3.2), where `sliderAdapter.position` is used to calculate `oldSign`, which results in `oldSign` being NaN, thus NaN being used as the `scrollOffset` parameter for `scrollbarAdapter.scrollTo()`. There the `roundToInt() `call on the `scrollOffset` triggers the actual Exception. The quickest fix for this would probably be a NaN check on sliderAdapter.position which stops the `scrollTo()` method even being called, for example by breaking from the while loop, just as the if in line 303 of the same file does.

**Comment by igordmn** [comment]
Should be fixed in 0.4.0-build190

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-6425">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
